pkg/credentialprovider/config.go

```diff
@@ -287,8 +287,8 @@ func decodeDockerConfigFieldAuth(field string) (username, password string, err e
 
 	// StdEncoding can only decode padded string
 	// RawStdEncoding can only decode unpadded string
-	// a string is correctly padded if and only if its length is a multiple of 4
-	if (len(field) % 4) == 0 {
+	field = strings.TrimRight(field, "\r\n")
+	if strings.HasSuffix(field, "=") {
```

Comment:
prefer `strings.HasSuffix(strings.TrimSpace(field), "=")`

Reply:
It a little bit less efficient, but probably more readable. Updated the PR and added the test that we don't trim too much.